src/btree/bt_read.c

```diff
@@ -125,17 +128,258 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+     * updates for the next key, and so on. In case if the birthmark record exists for that key,
+     * then insert birthmark record into the page.
+     *
+     * The search starts with minimum LAS key of the page until the maximum LAS key (both included).
+     * As we are going to instantiate only one record for a key and that too is a most recent one.
+     * To find out the most recent record for a key, we use the search_near cursor function with
+     * both maximum timestamp and transaction id as part of the key search. This search can return
+     * either the last record of the search key or the first record of the next key.
+     *
+     * Once we instantiate the most recent record for a key, the search continues to the next key.
+     * since we're pointing at the max timestamp and transaction id pairing for a given key, the
+     * next entry is guaranteed to be for the next key in the LAS and again set the search for the
+     * next key with maximum timestamp and transaction id.
+     */
+    cache->las_reader = true;
+    __wt_readlock(session, &cache->las_sweepwalk_lock);
+    cache->las_reader = false;
+    first_scan = true;
+    locked = true;
+    cursor->set_key(cursor, las_btree_id, &ref->page_las->min_las_key, WT_TS_MAX, WT_TXN_MAX);
+    for (; ret == 0; ret = cursor->next(cursor)) {
+        if (!first_scan) {
+            WT_ERR(
+              cursor->get_key(cursor, &las_btree_id, &next_las_key, &las_timestamp, &las_txnid));
+
+            /* Set the key to check with maximum timestamp and transaction id.*/
+            cursor->set_key(cursor, las_btree_id, &next_las_key, WT_TS_MAX, WT_TXN_MAX);
+        }
+        first_scan = false;
+
+        WT_ERR(cursor->search_near(cursor, &exact));
+        /*
+         * check whether the search_near returned the same key or not? In case if it returns next
+         * key, continue with the previous record.
+         */
+        if (exact > 0)
+            WT_ERR(cursor->prev(cursor));
+
+        WT_ERR(cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+
+        /* Stop before crossing over to the next btree */
+        if (las_btree_id != S2BT(session)->id)
+            break;
+
+        WT_ERR(__wt_compare(session, NULL, &las_key, &ref->page_las->min_las_key, &cmp));
+        /* we got a key that is less than the minimum key of the page */
+        if (cmp < 0)
+            break;
+
+        WT_ERR(__wt_compare(session, NULL, &las_key, &ref->page_las->max_las_key, &cmp));
+        /* we reached the end of the key of the LAS records related to the current page */
+        if (cmp > 0)
+            break;
+
+        instantiated_cnt++;
+        if (ref->page_las->birthmarks_cnt > 0) {
+            for (birthmarkp = ref->page_las->birthmarks + birthmark_cnt;
+                 birthmark_cnt < ref->page_las->birthmarks_cnt; birthmark_cnt++, birthmarkp++) {
```

Comment:
Continuation line indentation should be 2 spaces.

Reply:
It is the clang-format, that is doing it.